ERCS/erc-7694.md

```diff
@@ -0,0 +1,129 @@
+---
+eip: 7694
+title: Solana Storage Router
+description: Cross-chain storage router protocol incorporating storage router for Solana
+author: Avneet Singh (@sshmatrix), 0xc0de4c0ffee (@0xc0de4c0ffee)
+discussions-to: https://ethereum-magicians.org/t/erc-7694-solana-storage-router/19706
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-18
+---
+
+## Abstract
+The following standard is an extension to the cross-chain storage router protocol introducing the storage router for Solana blockchain. With this specification, any Ethereum L1 contract can defer a call to Solana blockchain as part of its core functionality, provided that the client is equipped to handle Solana transactions. It was previously possible to defer write and storage operations to other Ethereum L1 contracts, L2 contracts and off-chain databases, and this document extends that functionality to include alternative L1 chains. This standard allows Ethereum to utilise a broader range of cross-chain blockspaces, and puts it well ahead of other L1 ecosystems in its ability to interface with diverse cross-chain infrastructures.
```

Comment:
Your abstract is missing some technical meat. Summarize&mdash;in the document's abstract&mdash;**how** an Ethereum contract gets access to information on Solana. Does it use an oracle? Does the Ethereum contract contain a Solana light client implementation?

After reading the abstract, I should have a high-level understanding of how the proposal works (and not just what benefits it provides.)

Reply:
Hi @SamWilsn,

I have added a section describing the methods to read-in Solana data using CCIP-Read. See commit https://github.com/ethereum/ERCs/pull/390/commits/464937ed6b68c79d2a9c241340c86255c6e6a83e